UnityProject/Assets/Scripts/NPC/AI/MobAgent.cs

```diff
@@ -130,13 +130,26 @@ protected virtual void AgentServerStart()
 
 	void MonitorDecisionMaking()
 	{
-		if (health == null || integrity == null) return;
+		if (health == null && integrity == null) return;
 		// Only living mobs have health.  Some like the bots have integrity instead.
-		if (((health.IsDead || health.IsCrit || health.IsCardiacArrest || Pause)) ||
-			((integrity != null) && (integrity.integrity <= 0)))
+
+		if (Pause) return;
+
+		if (health != null) //Living mob
 		{
-			//can't do anything this NPC is not capable of movement
-			return;
+			if ((health.IsDead || health.IsCrit || health.IsCardiacArrest))
+			{
+				//can't do anything this NPC is not capable of movement
+				return;
+			}
+		}
+		else //Bot
+		{
+			if (integrity.integrity <= 0)
```

Comment:
integrity can be null which will give an NRE

Reply:
Line 133 should mean it can't get there if integrity is null